src/intervals/intervals.jl

```diff
@@ -106,9 +102,10 @@ end
 
 `interval(a, b)` checks whether [a, b] is a valid `Interval`, which is the case if `-∞ <= a <= b <= ∞`, using the (non-exported) `is_valid_interval` function. If so, then an `Interval(a, b)` object is returned; if not, then an error is thrown.
 """
-function interval(a::Real, b::Real)
+function interval(a::T, b::S) where {T<:Real, S<:Real}
     if !is_valid_interval(a, b)
-        throw(ArgumentError("`[$a, $b]` is not a valid interval. Need `a ≤ b` to construct `interval(a, b)`."))
+        @warn "Invalid input, empty interval is returned"
+        return emptyinterval(promote_type(T, S, Float64))
```

Comment:
This causes problems for invalid input of rationals, as the output would then be `Interval{Float64}` instead of `Interval{Rational}`, the `Float64` is there to avoid problems when both inputs are integers, but I guess that's not the smartest way to handle it

Reply:
What about defining `Interval(1//0, -1//0)` as the `nai` for rational intervals?

Reply:
This refers to my code review below?
Anyway, I think this would work because `nai` is only allowed as decorated interval, so you can distinguish `[1//0, -1//0]_trv` (the empty interval) from `[1//0, -1//0]_ill` (the NaI). This might require manually adjusting a couple of functions, but other then that it should work. Anyway, maybe this deserves its own issue/PR for further discussion?

Reply:
Sorry for misplacing my comment. Rather than manually requiring some adjustments, dispatching for `Interval{Rational{T}}` is also perhaps a way to proceed.

Reply:
I also opened an issue #462